File: output/nemotron-vs-27b/20260326-093607/agentic/workspaces/agentic_ts_metrics_contract_patch/nemotron_cascade_2_30b_a3b_iq4_xs/repeat-1/apps/game-client/src/state.ts
export type ClientState = {
  phase?: string;
  runtime?: Record<string, unknown>;
};

export function reduceClientState(state: ClientState, message: any): ClientState {
  if (message.type === "session.state") {
    return {
      ...state,
      phase: message.payload.phase,
      runtime: message.payload.runtime,
    };
  }
  if (message.type === "world.snapshot") {
    return {
      ...state,
      world: message.payload.world,
    };
  }
  return state;
}

File: output/omnicoder-vs-27b/20260325-omnicoder/agentic/workspaces/agentic_python_metrics_contract/omnicoder_9b_q8_0/repeat-2/client/state.py
from __future__ import annotations


def reduce_client_state(message: dict, state: dict) -> dict:
    if message["type"] == "session.state":
        payload = message["payload"]
        state["phase"] = payload["phase"]
        state["runtime"] = payload["runtime"]
    if message["type"] == "world.snapshot":
        state["world"] = message["payload"]["world"]
    return state


